src/scheduler.rs

```diff
@@ -20,20 +20,23 @@ pub struct Thread<'a> {
     pub st: &'a SymbolTable,
     pub current_stmt: StmtId,
     args: HashMap<&'a str, BitVecValue>,
+    fork_idx: usize,
```

Comment:
Can you add a description here about what a `fork_idx` is?
You can use a doc comment by stating with `///`